src/B0_data_input_json.py

```diff
@@ -30,7 +34,7 @@
 """
 
 
-def convert_special_types(a_dict, prev_key=None):
+def convert_from_json_to_special_types(a_dict, prev_key=None):
```

Comment:
Why call it "a_dict" if it is not clear that 'type(a_dict)==dict'? Would prefer renaming.

Reply:
It is a recursive function, it is still supposed to be used at first on json objects which are analogous to dict in python. Do you prefer `a_json`?